### DISCUSSION TITLE
Missing delay function for retry documentation.

### DISCUSSION THREAD

**Discussion by fudom** [ReactiveX/rxjs#7357] [Report issues other than bug]
It would be great if we find something about `delay` option with a function for the [retry](https://rxjs.dev/api/operators/retry) operator. Especially since [retryWhen](https://rxjs.dev/api/operators/retryWhen) is deprecated. The example should explain how to use the delay function and correct return value in conditional context like `retryWhen`. Currently there is only one example with simple `retry(2)`. And in the deprecated info of `retryWhen` where is only `retry({ delay: () => notify$ })`. Background: I need to migrate an Angular HTTP interceptor for conditional state error handling where no retry is needed. In my opinion it is very important to show such examples, especially when good things from RXJS suddenly become deprecated.

**Comment by blazkovicz** [comment]
Also want to add that documentation totally lacks counter description for delay function, internet examples equally suggest that it starts either from 0 or from 1. We had an issue here, only debug with console log showed that it starts with 1.
